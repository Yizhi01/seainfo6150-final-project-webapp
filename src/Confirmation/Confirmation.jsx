import React, {useState} from "react";
import ThankyouPage from "../ThankyouPage/ThankyouPage.jsx";
import styles from "./Confirmation.module.css";
import PropTypes from "prop-types";

const Confirmation = (props) => {
  const { title, description, serves, cookTime, difficulty, ingredients, instructions } = props;

  const [isConfirm, setConfirm] = useState();

  function handleClick(e) {
    e.preventDefault();
    const isConfirm =  true;
    setConfirm(isConfirm);
  }
  return (
    <div>
      {
      isConfirm ? (
        <ThankyouPage />
      ) : (
        <div className={styles.confirmContainer} >
          <p>Please confirm your recipe</p>
          <div className={styles.recipeDetail}>
            <h1 className={styles.recipeTitle}>{title}</h1>
            <hr/>
            <p>{description}</p>
            <hr/>
            <div>
              <p><span>SERVERS: </span>{serves}</p>
              <p><span>COOKS IN: </span>{cookTime}</p>
              <p><span>DIFFICULTY: </span>{difficulty}</p>
            </div>
            <hr />
            <div>
              <h2>Ingredients:</h2>
              <p>{ingredients}</p>
            </div>
            <hr />
            <div>
              <h2>Instructions:</h2>
              <p>{instructions}</p>
            </div>      
          </div>
          <div className={styles.action}>
            <input 
              className={styles.button} 
              type="submit" 
              value="Confirm recipe" 
              onClick={handleClick}/>
          </div>
        </div>
      )}
    </div>
  )
};

Confirmation.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  serves: PropTypes.string.isRequired,
  cookTime: PropTypes.string.isRequired,
  difficulty: PropTypes.string.isRequired,
  ingredients: PropTypes.string.isRequired,
  instructions: PropTypes.string.isRequired,
}

export default Confirmation;