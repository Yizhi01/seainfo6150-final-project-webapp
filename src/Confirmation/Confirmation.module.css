.confirmContainer {
  font-size: 1em;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  margin: 50px 100px;
  padding: 15px;
  border-radius: 10px;
  border: 7px solid #92817a;
}

.confirmContainer hr {
  width: 100%;
  border: 1px solid #bedbbb;
  margin: 15px 0;
}

.recipeDetail {
  margin: 20px 0 0;
  width: 100%;
}

.recipeTitle {
  color: #707070;
}

.action {
  margin: 20px 0 0 0;
  width: 100%;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
}

.button {
  font-size: 1em;
  margin: 30px 0;
  padding: 10px 20px;
  background-color: #59886b;
  color: #fff;
  border: 3px solid #59886b;
  border-radius: 5px;
}

.button:hover {
  background-color: #fff;
  color: #59886b;
  border: 3px solid #59886b;
}