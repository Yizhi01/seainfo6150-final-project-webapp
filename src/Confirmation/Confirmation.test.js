import React from "react";
import { render } from "@testing-library/react";
import Confirmation from "./Confirmation.jsx";

const testdata = {
  title: "cake",
  description: "cupcake",
  serves: "2",
  cookTime: "20 mins",
  difficulty: "easy",
  ingredients: "flour, sugar",
  instructions: "mix ingredients and bake cake",
};

it("renders the Confirmation component correctly", () => {
  const { container } = render(
    <Confirmation 
      title="cake"
      description="cupcake"
      serves="2"
      cookTime="20 mins"
      difficulty="easy"
      ingredients="flour, sugar"
      instructions="mix ingredients and bake cake" />
  );
  expect(container).toMatchSnapshot();
});