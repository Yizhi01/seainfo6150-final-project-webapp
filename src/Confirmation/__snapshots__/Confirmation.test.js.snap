// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Confirmation component correctly 1`] = `
<div>
  <div>
    <div
      class="confirmContainer"
    >
      <p>
        Please confirm your recipe
      </p>
      <div
        class="recipeDetail"
      >
        <h1
          class="recipeTitle"
        >
          cake
        </h1>
        <hr />
        <p>
          cupcake
        </p>
        <hr />
        <div>
          <p>
            <span>
              SERVERS: 
            </span>
            2
          </p>
          <p>
            <span>
              COOKS IN: 
            </span>
            20 mins
          </p>
          <p>
            <span>
              DIFFICULTY: 
            </span>
            easy
          </p>
        </div>
        <hr />
        <div>
          <h2>
            Ingredients:
          </h2>
          <p>
            flour, sugar
          </p>
        </div>
        <hr />
        <div>
          <h2>
            Instructions:
          </h2>
          <p>
            mix ingredients and bake cake
          </p>
        </div>
      </div>
      <div
        class="action"
      >
        <input
          class="button"
          type="submit"
          value="Confirm recipe"
        />
      </div>
    </div>
  </div>
</div>
`;
